### PR TITLE
Commenting and liking added

### PR DESCRIPTION
**Pull request by xxk3990**
To test:

Download and run the code. Run the react. Create a memory. Go into the db and verify it was added. Once the memory is added, it should take you to the list of memories with the one you just created added to it and all the necessary data showing. Try to like the post you added and add a comment to it. The number of likes should go up and the comment should appear in the comments list.  Add a second comment to the same post to verify both are appearing. That's all that has been implemented so far.

### CODE REVIEW

File: client/src/components/Memories.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+          })}
+        </section>
+      </div>
+    );
+  }
+ 
+}
+
+
+const MemoryTile = (props) => {
```
Comment (paulkogan) [inline client/src/components/Memories.js]: Can you break this out into its own file?

File: client/src/components/Memories.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      num_likes: memory.num_likes + 1,
+    }
+    const response = await handlePut(endpoint, requestBody)
+    if(response.status === 200 || response.status === 201) {
+      getMemories();
+    } else {
+      alert("post like failed.")
+    }
+  }
+  const submitComment = async (comment, setNewComment) => {
```
Comment (paulkogan) [inline client/src/components/Memories.js]: where is setNewComment defined or passed in from?

File: client/src/components/Memories.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        /> 
+        : null}
+      </section>
+    )
+  }
+}
+
+
+
+const ViewComments = (props) => {
```
Comment (paulkogan) [inline client/src/components/Memories.js]: same here. Should be a standalone file

File: client/src/components/Navbar.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import logo from '../logo.svg';
+import '../styles/navbar.css'
+import React  from 'react';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+export default function Navbar() {
+    return (
+        <div className='Navbar'>
+            <section className='nav-loggedin'>
+                <ul className = "nav-links">
```
Comment (paulkogan) [inline client/src/components/Navbar.js:10]: Link to Landing page?

File: client/src/services/requests-service.js [hunk 5]
```diff
@@ -20,4 +20,18 @@ export const handlePost = (endpoint, body) => {
         body: JSON.stringify(body)
     }
     return fetch(url, requestParams)
+}
+
+export const handlePut = async (endpoint, body) => {
```
Comment (paulkogan) [inline client/src/services/requests-service.js:18]: You should use Axios for FE requests. More reliable with better error handling than fetch. Its kind of a standard.

File: server/controllers/memories-controller.js [hunk 6]
```diff
@@ -1,27 +1,133 @@
...
 const getMemories = async (req, res) => {
-    const memories = await models.Memory.findAll();
-    if(memories.length !== 0) {
-        return res.status(200).send(memories)
+    const memories = await models.Memory.findAll({
+        raw: true
+    });
+    if (memories.length !== 0) {
+        const posts = [...memories]
+        for (let i = 0; i < posts.length; i++) {
```
Comment (paulkogan) [inline server/controllers/memories-controller.js:12]: So yea, dont do this. This is exactly what "include" is for. 
If you properly associate memories and comments, you should automatically get all the comments for a memory with "include"

File: client/src/components/Memories.js (hunk 2 above)
Comment (paulkogan) [reply client/src/components/Memories.js]: and where is comment object being defined?

**Review by paulkogan — COMMENTED** [review]
Good work.
Left a few comments to fix up.
Let me know when done and I will approve

File: client/src/components/Memories.js (hunk 2 above)
Comment (xxk3990) [reply client/src/components/Memories.js]: The comment object is passed in from the submit comment method in the ViewComments mini-component

File: server/controllers/memories-controller.js (hunk 6 above)
Comment (xxk3990) [reply server/controllers/memories-controller.js:12]: If I use include I can't get the commenter's name as that comes from the user model, so I still have to get the user based on the comment's user_uuid.

File: client/src/services/requests-service.js (hunk 5 above)
Comment (xxk3990) [reply client/src/services/requests-service.js:18]: Done!

File: client/src/components/Navbar.js (hunk 4 above)
Comment (xxk3990) [reply client/src/components/Navbar.js:10]: Added!

File: client/src/components/Memories.js (hunk 1 above)
Comment (xxk3990) [reply client/src/components/Memories.js]: This is part of the memories component. It's just in a new method so there's one for each individual memory

File: client/src/components/Memories.js (hunk 3 above)
Comment (xxk3990) [reply client/src/components/Memories.js:134]: Should child components get their own file?

File: client/src/components/Memories.js (hunk 1 above)
Comment (xxk3990) [reply client/src/components/Memories.js]: Never mind! I moved both to a new file!

File: client/src/components/Memories.js (hunk 3 above)
Comment (xxk3990) [reply client/src/components/Memories.js]: Done!

File: server/controllers/memories-controller.js (hunk 6 above)
Comment (xxk3990) [reply server/controllers/memories-controller.js:12]: I can use include to get the comment text, but in order to get the user who posted the comment's display_name I need the comment's user_uuid, which I don't think I can get from the memory and comment association but I could be wrong!

**Review by paulkogan — APPROVED** [review]
looking good!

**Comment by xxk3990** [comment]
> looking good!

Thank you!

**Pull request merged**